parastell/utils.py

```diff
@@ -56,124 +58,63 @@ def expand_ang_list(ang_list, num_ang):
-    'mat_tag': 'magnets',
-    'export_mesh': False,
-    'mesh_filename': 'magnet_mesh',
-    'export_dir': ''
-}
-
-# source_def (dict): dictionary of source mesh parameters including
-#     {
-#         'scale': a scaling factor between the units of VMEC and [cm]
-#             (double, defaults to m2cm = 100).
-#         'filename': name of H5M output file, excluding '.h5m' extension
-#             (str, defaults to 'source_mesh').
-#         'export_dir': directory to which to export H5M output file
-#             (str, defaults to empty string).
-#     }
-source_def = {
-    'scale': m2cm,
-    'filename': 'source_mesh',
-    'export_dir': ''
-}
-
-# dagmc_export_def (dict): dictionary of DAGMC export parameters including
-#     {
-#         'skip_imprint': choose whether to imprint and merge all in
-#             Coreform Cubit or to merge surfaces based on import order
-#             and geometry information (bool).
-#         'legacy_faceting': choose legacy or native faceting for DAGMC
-#             export (bool).
-#         'faceting_tolerance': maximum distance a facet may be from
-#             surface of CAD representation for DAGMC export (double).
-#         'length_tolerance': maximum length of facet edge for DAGMC
-#             export (double).
-#         'normal_tolerance': maximum change in angle between normal
-#             vector of adjacent facets (double).
-#         'anisotropic_ratio': controls edge length ratio of elements
-#             (double).
-#         'deviation_angle': controls deviation angle of facet from
-#             surface (i.e., lesser deviation angle results in more
-#             elements in areas with greater curvature) (double).
-#         'filename': name of DAGMC output file, excluding '.h5m'
-#             extension (str, defaults to 'dagmc').
-#         'export_dir': directory to which to export DAGMC output file
-#             (str, defaults to empty string).
-#             }
-dagmc_export_def = {
-    'skip_imprint': False,
-    'legacy_faceting': True,
-    'faceting_tolerance': None,
-    'length_tolerance': None,
-    'normal_tolerance': None,
-    'anisotropic_ratio': 100,
-    'deviation_angle': 5,
-    'filename': 'dagmc',
-    'export_dir': ''
-}
+
+def construct_kwargs_from_dict(
+    dict, allowed_kwargs, all_kwargs, fn_name=None, logger=None
+):
+    """Constructs a dictionary of keyword arguments with corresponding values
+    for a class method from an input dictionary based on a list of allowable
+    keyword argument argument names. Conditionally raises an exception if the
+    user flags that the supplied list of allowable arguments should represent
+    all keys present in the input dictionary of arguments.
+
+    Arguments:
+        dict (dict): dictionary of arguments and corresponding values.
+        allowed_kwargs (list of str): list of allowed keyword argument names.
+        all_kwargs (bool): flag to indicate whether 'allowed_kwargs' should
```

Comment:
Should this default to `False`?

Reply:
I'm not opposed to setting a default for it; would be one less argument to add in some of the function calls, and defaulting to `False` seems appropriate.